sonar-helm-for-iac/build.gradle.kts

```diff
@@ -19,13 +21,55 @@ artifacts.add(goBinaries.name, goBinariesJar)
 
 val isCi: Boolean = System.getenv("CI")?.equals("true") ?: false
 
+tasks.register<Download>("downloadProtocGenGo") {
```

Comment:
It looks like the task is not cacheable. My output by 2nd run of `./gradlew build`:
```
> Task :its:ruling:spotlessApply UP-TO-DATE

> Task :sonar-helm-for-iac:downloadProtocGenGo
Download https://github.com/protocolbuffers/protobuf-go/releases/download/v1.34.1/protoc-gen-go.v1.34.1.darwin.arm64.tar.gz

> Task :sonar-helm-for-iac:generateProto UP-TO-DATE
```

Maybe this can be improved?

Reply:
Looks like it was because of the `doLast`, which is a custom action that affects caching. After extracting it into a separate task, I see
```
> Task :sonar-helm-for-iac:downloadProtocGenGo UP-TO-DATE
Download https://github.com/protocolbuffers/protobuf-go/releases/download/v1.34.1/protoc-gen-go.v1.34.1.linux.amd64.tar.gz

> Task :sonar-helm-for-iac:extractProtocGenGo UP-TO-DATE
```